frontend/src/metabase/visualizations/visualizations/SmartScalar.jsx

```diff
@@ -93,8 +94,8 @@ export default class Smart extends React.Component {
       rawSeries,
     } = this.props;
 
-    const metricIndex = 1;
-    const dimensionIndex = 0;
+    const metricIndex = _.findIndex(cols, col => !isDate(col));
+    const dimensionIndex = _.findIndex(cols, isDate);
```

Comment:
Would this work without underscore.js? 🤔

```suggestion
    const metricIndex = cols.findIndex(col => !isDate(col));
    const dimensionIndex = cols.findIndex(col => isDate(col));
```

Reply:
Seemed to work locally for me.